bin/_mocha

```diff
@@ -7,646 +7,35 @@
-program.compilers.forEach(c => {
-  const idx = c.indexOf(':');
-  const ext = c.slice(0, idx);
-  let mod = c.slice(idx + 1);
-
-  if (mod[0] === '.') {
-    mod = join(process.cwd(), mod);
-  }
-  require(mod);
-  extensions.push(ext);
-  program.watchExtensions.push(ext);
-});
-
-// requires
-
-requires.forEach(mod => {
-  require(mod);
-});
-
-// interface
-
-mocha.ui(program.ui);
-
-// args
-
-const args = program.args;
-
-// default files to test/*.{js,coffee}
-
-if (!args.length) {
-  args.push('test');
-}
-
-args.forEach(arg => {
-  let newFiles;
-  try {
-    newFiles = utils.lookupFiles(arg, extensions, program.recursive);
-  } catch (err) {
-    if (err.message.indexOf('cannot resolve path') === 0) {
-      console.error(
-        `Warning: Could not find any test files matching pattern: ${arg}`
-      );
-      return;
-    }
-
-    throw err;
-  }
-
-  if (typeof newFiles !== 'undefined') {
-    if (typeof newFiles === 'string') {
-      newFiles = [newFiles];
-    }
-    newFiles = newFiles.filter(fileName =>
-      program.exclude.every(pattern => !minimatch(fileName, pattern))
-    );
-  }
-
-  files = files.concat(newFiles);
-});
-
-if (!files.length) {
-  console.error('No test files found');
-  process.exit(1);
-}
-
-// resolve
-let fileArgs = program.file.map(path => resolve(path));
-files = files.map(path => resolve(path));
-
-if (program.sort) {
-  files.sort();
-}
-
-// add files given through --file to be ran first
-files = fileArgs.concat(files);
-
-// --watch
-
-let runner;
-let loadAndRun;
-let purge;
-let rerun;
-
-if (program.watch) {
-  console.log();
-  hideCursor();
-  process.on('SIGINT', () => {
-    showCursor();
-    console.log('\n');
-    process.exit(130);
-  });
-
-  const watchFiles = utils.files(cwd, program.watchExtensions);
-  let runAgain = false;
-
-  loadAndRun = () => {
-    try {
-      mocha.files = files;
-      runAgain = false;
-      runner = mocha.run(() => {
-        runner = null;
-        if (runAgain) {
-          rerun();
-        }
-      });
-    } catch (e) {
-      console.log(e.stack);
-    }
-  };
-
-  purge = () => {
-    watchFiles.forEach(file => {
-      delete require.cache[file];
-    });
-  };
-
-  loadAndRun();
-
-  rerun = () => {
-    purge();
-    stop();
-    if (!program.grep) {
-      mocha.grep(null);
-    }
-    mocha.suite = mocha.suite.clone();
-    mocha.suite.ctx = new Mocha.Context();
-    mocha.ui(program.ui);
-    loadAndRun();
-  };
-
-  utils.watch(watchFiles, () => {
-    runAgain = true;
-    if (runner) {
-      runner.abort();
-    } else {
-      rerun();
-    }
-  });
-} else {
-  // load
-
-  mocha.files = files;
-  runner = mocha.run(program.exit ? exit : exitLater);
-}
-
-process.on('SIGINT', () => {
-  runner.abort();
-
-  // This is a hack:
-  // Instead of `process.exit(130)`, set runner.failures to 130 (exit code for SIGINT)
-  // The amount of failures will be emitted as error code later
-  runner.failures = 130;
-});
```

Comment:
Even sweeter would be to reduce "bin/_mocha" to something like [this](https://github.com/cybertk/abao/blob/master/bin/abao)...

Reply:
- [ ] destroy `_mocha` more

Reply:
Well, if gets down to just a couple lines, all coverage is basically tracked within library.